### PR TITLE
Move authorization list and access list padding from importer to rest

### PR DESCRIPTION
**Pull request by bilyana-gospodinova**
**Description**:
The zero left-padding of some fields in Ethereum transaction's access list and authorization list is now moved from importer to the rest module. The rest endpoints will return the same result to the users. This shift ensures the integrity of the data in the importer as early padding changes the transaction hash. Now the transaction hash will be intact and the overall result from the endpoints will be the same.

MN-76

### CODE REVIEW

**Review by IvanKavaldzhiev — DISMISSED** [review]
LGTM

**Review by martingeorgiev1 — DISMISSED** [review]
LGTM

File: rest/utils.js [hunk 1]
```diff
@@ -1106,6 +1106,53 @@ const toHexStringNonQuantity = (byteArray) => {
   return toHexString(byteArray, true, 2);
 };
 
+const padHex = (hex, padLength) => {
```
Comment (steven-sheehy) [inline rest/utils.js]: Why can't we use the existing `toHexString(padLength = 40)`?

File: rest/utils.js [hunk 2]
```diff
@@ -1106,6 +1106,53 @@ const toHexStringNonQuantity = (byteArray) => {
...
+  }
+  return constants.HEX_PREFIX + stripHexPrefix(hex).padStart(padLength, '0');
+};
+
+/**
+ * Left-pads access list address to 20 bytes and storage keys to 32 bytes for REST responses.
+ * @param {Array|null|undefined} accessList
+ * @return {Array}
+ */
+const padAccessList = (accessList) => {
```
Comment (steven-sheehy) [inline rest/utils.js]: Can't these two methods live in the parent class `ContractResultViewModel`?

File: rest/utils.js [hunk 3]
```diff
@@ -1106,6 +1106,53 @@ const toHexStringNonQuantity = (byteArray) => {
...
+    return accessList ?? [];
+  }
+
+  return accessList.map((entry) => {
+    const storageKeys = entry.storage_keys ?? entry.storageKeys;
+    return {
+      ...entry,
+      address: padHex(entry.address, constants.EVM_ADDRESS_LENGTH * 2),
+      storage_keys: Array.isArray(storageKeys)
+        ? storageKeys.map((key) => padHex(key, constants.ETH_HASH_LENGTH * 2))
```
Comment (steven-sheehy) [inline rest/utils.js]: It's only coincidence that the hash length matches the slot length. We should add a separate constant and use it in `slotPattern` as well.

File: rest/utils.js [hunk 4]
```diff
@@ -1106,6 +1106,53 @@ const toHexStringNonQuantity = (byteArray) => {
...
+        : storageKeys,
```
Comment (steven-sheehy) [inline rest/utils.js]: Won't this return null if storageKeys is not an array? We should default to [] if null.

File: rest/viewmodel/contractResultViewModel.js [hunk 5]
```diff
@@ -28,7 +28,7 @@ class ContractResultViewModel {
       : contractId.toEvmAddress();
     this.amount = contractResult.amount;
     if (config.response.enableDelegationAddress && !isEmpty(contractResult.authorizationList)) {
-      this.authorization_list = contractResult.authorizationList || [];
+      this.authorization_list = padAuthorizationList(contractResult.authorizationList || []);
```
Comment (steven-sheehy) [inline rest/viewmodel/contractResultViewModel.js]: Shouldn't need the ` || []` since the `padAuthorizationList` method should handle null.

File: rest/viewmodel/contractResultDetailsViewModel.js [hunk 6]
```diff
@@ -88,9 +88,9 @@ class ContractResultDetailsViewModel extends ContractResultViewModel {
     this.nonce = null;
 
     if (!isNil(ethTransaction)) {
-      this.access_list = ethTransaction.accessList ?? [];
+      this.access_list = utils.padAccessList(ethTransaction.accessList ?? []);
       if (config.response.enableDelegationAddress) {
-        this.authorization_list = ethTransaction.authorizationList ?? [];
+        this.authorization_list = utils.padAuthorizationList(ethTransaction.authorizationList ?? []);
```
Comment (steven-sheehy) [inline rest/viewmodel/contractResultDetailsViewModel.js]: Shouldn't need the ` || []` since the `padAuthorizationList` method should handle null.

File: rest/__tests__/utils.test.js [hunk 7]
```diff
@@ -1294,6 +1294,122 @@ describe('Utils toHexStringNonQuantity and toHexStringQuantity tests', () => {
...
+        storage_keys: [paddedStorageKey],
+      },
+    ]);
+  });
+
+  test('leaves already padded values unchanged', () => {
+    const accessList = [
+      {
+        address: paddedAddress,
+        storage_keys: [paddedStorageKey],
```
Comment (steven-sheehy) [inline rest/__tests__/utils.test.js]: Add a test for null and empty storage_keys

File: importer/src/main/java/org/hiero/mirror/importer/parser/record/ethereum/Eip7702EthereumTransactionParser.java [hunk 8]
```diff
@@ -102,20 +101,16 @@ private List<Authorization> parseAuthorizationList(RLPItem authorizationListItem
 
             final var hexFormat = HexFormat.of();
             final var chainId = tuple.get(0).data();
-            final var r = tuple.get(4).data();
-            final var s = tuple.get(5).data();
```
Comment (steven-sheehy) [inline importer/src/main/java/org/hiero/mirror/importer/parser/record/ethereum/Eip7702EthereumTransactionParser.java:106]: Should limit the number of authorizations.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: rest/utils.js (hunk 1 above)
Comment (bilyana-gospodinova) [reply rest/utils.js]: We can. `toHexString(...)` expects an input without the "0x" prefix, but it can be used like this, for example:
`toHexString(stripHexPrefix(entry.address), true, constants.EVM_ADDRESS_LENGTH * 2)`.

Comment (steven-sheehy) [reply rest/utils.js]: Should just update it to only add prefix if it's not already there instead of always stripping and re-adding.

Comment (bilyana-gospodinova) [reply rest/utils.js]: Updated

File: rest/utils.js (hunk 2 above)
Comment (bilyana-gospodinova) [reply rest/utils.js]: Moved

File: rest/utils.js (hunk 3 above)
Comment (bilyana-gospodinova) [reply rest/utils.js]: Introduced a new constant.

File: rest/utils.js (hunk 4 above)
Comment (bilyana-gospodinova) [reply rest/utils.js]: Addressed

File: rest/viewmodel/contractResultViewModel.js (hunk 5 above)
Comment (bilyana-gospodinova) [reply rest/viewmodel/contractResultViewModel.js]: Addressed

File: rest/viewmodel/contractResultDetailsViewModel.js (hunk 6 above)
Comment (bilyana-gospodinova) [reply rest/viewmodel/contractResultDetailsViewModel.js]: Addressed

File: rest/__tests__/utils.test.js (hunk 7 above)
Comment (bilyana-gospodinova) [reply rest/__tests__/utils.test.js]: Addressed

File: importer/src/main/java/org/hiero/mirror/importer/parser/record/ethereum/Eip7702EthereumTransactionParser.java (hunk 8 above)
Comment (bilyana-gospodinova) [reply importer/src/main/java/org/hiero/mirror/importer/parser/record/ethereum/Eip7702EthereumTransactionParser.java:106]: Addressed

**Review by steven-sheehy — APPROVED** [review]
